ReactWindows/ReactNative.Shared/Views/TextInput/ReactTextBox.cs

```diff
@@ -104,6 +134,46 @@ public int IncrementEventCount()
             return Interlocked.Increment(ref _eventCount);
         }
 
+        protected override void OnApplyTemplate()
+        {
+            base.OnApplyTemplate();
+            if (DeleteButton == null)
```

Comment:
This will prevent the logic from kicking in in case there are multiple calls to OnApplyTemplate - the code will go through the first time but subsequent calls (for which we introduced unregistering will never go through. I guess the check for DeleteButton==null should be removed

Reply:
It will also be necessary to put the un-registration before the rest i.e.

if (DeleteButton != null)
{
   DeleteButton.Unregister......
}

DeleteButton= (Button)GetTemplateChild("")
DeleteButtonVisibiltiyToken = .....




---
In reply to: [198667759](https://github.com/Microsoft/react-native-windows/pull/1874#discussion_r198667759) [](ancestors = 198667759)